### DISCUSSION TITLE
Sending Complex Data Structures From React To Unity

### DISCUSSION THREAD

**Discussion by hedgehoghash** [jeffreylanters/react-unity-webgl#267] [Question and Answer]
Is there a way to send a complex data type other than a string, number, or boolean from React to Unity? Is the best way to do this sending a JSON string from React to Unity and then Unity would deserialize that JSON into a class?

**Comment by jeffreylanters** [comment]
Hi Tommy. Unity only allows [a set number of types](https://github.com/jeffreylanters/react-unity-webgl#javascript-to-unityscript-types) to be used as parameters when sending messages back to Unity. So indeed, JSON would be the best approach.
